### PR TITLE
chore(vote) change webpack.vote to benefit from latest commits and in…

### PR DESCRIPTION
**Pull request by EugeneHlushko**
…stall it from git vs npm (we dont have rights to publish)

- mostly said in title, don't have control over package on npm. As we move it will now be fetched from git with `^v0.3.0`

### CODE REVIEW

**Comment by montogeek** [comment]
Who has control?

**Comment by EugeneHlushko** [comment]
I need to fix export of the library, `yarn run start` locally works but production build fails with 
```
ERROR in TypeError: Super expression must either be null or a function, not undefined
    at evalmachine.<anonymous>:104:18230
    at evalmachine.<anonymous>:104:18313
    at Object.n.exports (evalmachine.<anonymous>:104:19429)
    at s (evalmachine.<anonymous>:104:15848)
    at Object.n.exports (evalmachine.<anonymous>:104:40516)
    at s (evalmachine.<anonymous>:104:15848)
    at Object.n.exports (evalmachine.<anonymous>:104:24780)
    at s (evalmachine.<anonymous>:104:15848)
    at n.exports (evalmachine.<anonymous>:104:16648)
    at Object.<anonymous> (evalmachine.<anonymous>:104:16657)
```

Which can have multiple causes, i guess it occured during the webpack upgrade v3 -> v4 in the vote app repo

**Comment by anshumanv** [comment]
> Which can have multiple causes, i guess it occured during the webpack upgrade v3 -> v4 in the vote app repo

Hmm that's unusual, worked well for me when I was working with it. Should I take a look at it?

**Comment by EugeneHlushko** [comment]
@anshumanv thank but i know what it is, its the SSG run `target: node` that's killing it, i have a fix for it already but want to upgrade react to 16 first

**Comment by anshumanv** [comment]
> @anshumanv thank but i know what it is, its the SSG run target: node that's killing it, i have a fix for it already but want to upgrade react to 16 first

Awesome, please let me know if I can help somewhere. :)

**Comment by bebraw** [comment]
@montogeek I think it’s behind Tobias as he built the voting bits.

**Comment by EugeneHlushko** [comment]
I think its on @skipjack you can see this on npm: https://www.npmjs.com/package/webpack.vote

File: package.json [hunk 1]
```diff
@@ -139,10 +139,10 @@
     "static-site-generator-webpack-plugin": "^3.4.1",
     "style-loader": "^0.18.2",
     "tap-spot": "^1.1.1",
-    "terser-webpack-plugin": "^1.2.3",
+    "terser-webpack-plugin": "^2.3.5",
     "through2": "^2.0.3",
     "uglifyjs-webpack-plugin": "^1.1.6",
-    "webpack": "^4.30.0",
+    "webpack": "^4.42.0",
```
Comment (EugeneHlushko) [inline package.json]: Terser was bugged, fails came not from target but from minimization.

File: src/components/Vote/Vote.jsx [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
-);
-
-export default Vote;
+export default class Vote extends React.Component {
+  state = {
+    VoteApp: null
+  }
+  componentDidMount() {
+    if (window && !window.__ssgrun) {
+      import('webpack.vote').then(VoteApp => {
```
Comment (EugeneHlushko) [inline src/components/Vote/Vote.jsx]: it wasnt rendered server side anyway so we only import this client side dynamically

**Comment by EugeneHlushko** [comment]
@montogeek want to have another look ?

File: package.json [hunk 3]
```diff
@@ -139,10 +139,10 @@
     "static-site-generator-webpack-plugin": "^3.4.1",
     "style-loader": "^0.18.2",
     "tap-spot": "^1.1.1",
-    "terser-webpack-plugin": "^1.2.3",
+    "terser-webpack-plugin": "^2.3.5",
```
Comment (montogeek) [inline package.json]: It is ok to remove it, right? It is not longer used.

Comment (EugeneHlushko) [reply package.json]: @montogeek it is used, actually bundlesize growth and build failure yesterday due to removing it. Almost 2x gzipped without passing it directly as a minimizer as i wrote above, i was sure its safe to remove it and simply do `optimization.minimize: true`.

Whats interesting in lockfile for webpack 4.42 1.4 version of it used, so the one we pass, is actually next major v

**Comment by skipjack** [comment]
> I think its on @skipjack you can see this on npm: https://www.npmjs.com/package/webpack.vote

You guys need me to cut / publish a new release? Also, maybe I can add one of your npm accounts to that package so you can publish yourself next time?

**Comment by EugeneHlushko** [comment]
Its fine we can use github url now so no worries

**Pull request merged**